drivers/storage/portworx/component/monitoring.go

```diff
@@ -31,14 +39,61 @@ const (
 	PxBackupServiceMonitor = "px-backup"
 	// PxPrometheusRule name of the prometheus rule object for Portworx
 	PxPrometheusRule = "portworx"
+	// GrafanaDeploymentName is the name for the operator managed grafana
+	GrafanaDeploymentName = "grafana"
+	// GrafanaContainerName is the name for the grafana container
+	GrafanaContainerName = "grafana"
 
+	grafanaDefaultCPU    = "0.1"
+	grafanaDefaultMemory = "0.1"
+	grafanaPort          = 3000
+	grafanaLoginEndpoint = "/login"
 	pxPrometheusRuleFile = "portworx-prometheus-rule.yaml"
 )
 
+var (
+	grafanaDeploymentVolumes = []corev1.VolumeSpec{
+		{
+			Name:      "grafana-source-config",
+			MountPath: "/etc/grafana/provisioning/datasources",
```

Comment:
When configuring the datasource, we will have to add our operator deployed prometheus to it. We should probably only create the dashboard configmap or even grafana only if our prometheus is deployed. Wdyt?